### PR TITLE
ARC-1661: fix repo id collisions

### PR DESCRIPTION
**Pull request by bgvozdev**
**What's in this PR?**
There are only 4 cases in which we send RepoIds to Jira:
 - bulk submit commits
 - bulk submit branches
 - bulk submit PRs
 - commits associations
(see details https://atlassian.slack.com/archives/CFG3W7TKJ/p1665019072929499?thread_ts=1665018583.114159&cid=CFG3W7TKJ )

And there are only 3 places where we use them to delete data in Jira:
 - delete branch
 - delete PR
 - delete repo

**Why**
https://hello.atlassian.net/wiki/spaces/OTFS/pages/1927359889/GHES+vs+Repository+IDs

**Added feature flags**
https://switcheroo.atlassian.com/changes/github-for-jira/use-repo-id-transformer

**Affected issues**  
ARC-1661

**How has this been tested?**  
Manually, staging

NOTES:
 - when review, make sure to turn off whitespaces in diffs, or you will suffer mightily
 - I'm setting up a session for us to walk through

### CODE REVIEW

File: src/github/branch.ts [hunk 1]
```diff
@@ -80,7 +81,7 @@ export const deleteBranchWebhookHandler = async (context: WebhookContext, jiraCl
 	context.log.info({ prRef: context.payload.ref }, `Deleting branch for repo ${context.payload.repository?.id}`);
 
 	const jiraResponse = await jiraClient.devinfo.branch.delete(
-		`${payload.repository?.id}`,
+		transformRepositoryId(payload.repository?.id, context.gitHubAppConfig?.gitHubBaseUrl),
```
Comment (bgvozdev) [inline src/github/branch.ts:84]: this is the invocation to delete the branch, called on a webhook

File: src/github/pull-request.ts [hunk 2]
```diff
@@ -65,7 +66,7 @@ export const pullRequestWebhookHandler = async (context: WebhookContext, jiraCli
 			);
 
 			await jiraClient.devinfo.pullRequest.delete(
-				repositoryId,
+				transformRepositoryId(repositoryId, context.gitHubAppConfig?.gitHubBaseUrl),
```
Comment (bgvozdev) [inline src/github/pull-request.ts:69]: This is an invocation to delete the repo

File: src/github/repository.ts [hunk 3]
```diff
@@ -9,7 +10,7 @@ export const deleteRepositoryWebhookHandler = async (context: WebhookContext, ji
 	context.log.info(`Deleting dev info for repo ${context.payload.repository?.id}`);
 
 	const jiraResponse = await jiraClient.devinfo.repository.delete(
-		context.payload.repository?.id
+		transformRepositoryId(context.payload.repository?.id, context.gitHubAppConfig?.gitHubBaseUrl)
```
Comment (bgvozdev) [inline src/github/repository.ts:13]: This is an invocation to delete a Repo

File: src/interfaces/jira.ts [hunk 4]
```diff
@@ -8,6 +10,13 @@ interface JiraPullRequestRef {
 	uri: string;
 }
 
+export interface BulkSubmitRepositoryInfo {
```
Comment (bgvozdev) [inline src/interfaces/jira.ts:13]: 3 payloads have this: bulk submit commits, branches and PRs. This tiny refactoring allowed me to create `transform-repository`

File: src/jira/client/jira-client.ts [hunk 5]
```diff
@@ -143,15 +144,15 @@ export const getJiraClient = async (
 		},
 		devinfo: {
 			branch: {
-				delete: (repositoryId: string, branchRef: string) =>
+				delete: (transformedRepositoryId: TransformedRepositoryId, branchRef: string) =>
```
Comment (bgvozdev) [inline src/jira/client/jira-client.ts:147]: drag-n-drop replacement (you can highlight the new values with Control + F + "transformedRepositoryId" - no typos)

File: src/routes/github/webhook/webhook-context.ts [hunk 6]
```diff
@@ -21,9 +20,8 @@ export class WebhookContext<E = any> {
 	sentry?: Sentry.Hub;
 	timedout?: number;
 	webhookReceived?: number;
-	gitHubAppConfig?: GitHubAppConfig;
-	// Todo: delete github once all references to github has been removed. Found some reference behind FF
-	github: GitHubAPI;
```
Comment (bgvozdev) [inline src/routes/github/webhook/webhook-context.ts:26]: Doesn't look we use it anywhere

File: src/sync/transforms/branch.ts [hunk 7]
```diff
@@ -95,11 +101,8 @@ export const transformBranches = async (payload) => {
 	}
 
 	return {
+		... transformRepositoryDevInfoBulk(payload.repository, gitHubBaseUrl),
```
Comment (bgvozdev) [inline src/sync/transforms/branch.ts:104]: Drag-n-drop replacement (with exception of changing `repository.name` to `repository.full_name` - this is what we have in other places), not feature-flagging
image

File: src/sync/transforms/pull-request.ts [hunk 8]
```diff
@@ -27,8 +35,7 @@ export const transformPullRequest =  async (payload: Payload, prDetails: Octokit
 	}
 
 	return {
-		id: repository.id.toString(),
-		name: repository.full_name,
+		... transformRepositoryDevInfoBulk(repository, gitHubBaseUrl),
```
Comment (bgvozdev) [inline src/sync/transforms/pull-request.ts:38]: Drag-n-drop replacement (with exception of changing repository.name to repository.full_name - this is what we have in other places), not feature-flagging
image
(on the left - removed lines, on the right - the mapping inside the function)

File: src/transforms/transform-branch.ts [hunk 9]
```diff
@@ -39,11 +40,9 @@ export const transformBranch = async (github: GitHubInstallationClient, webhookP
 	}
 
 	try {
-		const lastCommit = await getLastCommit(github, webhookPayload, issueKeys);
+		const lastCommit = await getLastCommit(gitHubInstallationClient, webhookPayload, issueKeys);
 		return {
-			id: repository.id.toString(),
-			name: repository.full_name,
-			url: repository.html_url,
+			... transformRepositoryDevInfoBulk(repository, gitHubInstallationClient.baseUrl),
```
Comment (bgvozdev) [inline src/transforms/transform-branch.ts:45]: Drag-n-drop replacement (with exception of changing repository.name to repository.full_name - this is what we have in other places), not feature-flagging
image
(on the left - removed lines, on the right - the mapping inside the function)

File: src/transforms/transform-commit.ts [hunk 10]
```diff
@@ -34,10 +40,7 @@ export const transformCommit = (payload): JiraCommitData | undefined => {
 	}
 
 	return {
-		commits: commits,
-		id: payload.repository.id.toString(),
-		name: payload.repository.full_name,
-		url: payload.repository.html_url,
-		updateSequenceId: Date.now()
+		... transformRepositoryDevInfoBulk(payload.repository, gitHubBaseUrl),
```
Comment (bgvozdev) [inline src/transforms/transform-commit.ts:43]: Drag-n-drop replacement (with exception of changing repository.name to repository.full_name - this is what we have in other places), not feature-flagging
image
(on the left - removed lines, on the right - the mapping inside the function)

File: src/github/repository.test.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+		when(booleanFlag).calledWith(
+			BooleanFlags.USE_REPO_ID_TRANSFORMER,
+			expect.anything()
+		).mockResolvedValue(true);
+	});
+
+	it('should call delete repository endpoint for server', async () => {
+
+		const builderResult = await new DatabaseStateCreator()
```
[the remark this answers is not in the document]
Comment (bgvozdev) [reply src/github/repository.test.ts:28]: We need it for the webhook context... not sure if the data is actually used or not

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**